coremltools/converters/mil/mil/passes/defs/optimize_normalization.py

```diff
@@ -25,6 +25,130 @@ class fuse_layernorm_or_instancenorm(AbstractGraphPass):
     A graph optimization pass on PyMIL to detect and fuse several variants of ``layer_norm`` or
     ``instance_norm``. Pattern 1 corresponds to either ``layer_norm`` or ``instance_norm``. Patterns 2-4
     are ``instance_norm``.
+
+    `Pattern 1`
+    
+    Identify the pattern:
+
+    ``y = gamma * (x - mean) / sqrt(variance + epsilon) + beta``
+
+    ``y = x * [gamma * rsqrt(variance + eps)] + (beta - mean * [gamma * rsqrt(variance + eps)])``
+    
+    .. code-block::
+
+            x --> reduce_mean --> sub --> square --> reduce_mean --> add(epsilon) --> rsqrt
+            |             |        ^                                                    |
+            |             |        |                                                    V
+            |-----------------------                                              mul (gamma)
+            |             |                                                           |
+            |             |                                                   --------|---------
+            |             |                                                   |                |
+            |             |                                                   |                V
+            |             |---------------------------------------------------------------->  mul
+            |                                                                 |                |
+            |                                                                 V                |
+            |--------------------------------------------------------------> mul               |
+                                                                              |                V
+                                                                              |              sub (beta) --> add --> [...]
+                                                                              |                              ^
+                                                                              |-------------------------------
+
+    This pattern corresponds to either ``layer_norm`` or ``instance_norm``.
+
+    It is ``instance_norm`` if all of the following are true:
+        - ``input`` is rank 4.
+        - ``axes`` of ``reduce_mean`` is ``[-2, -1]`` or ``[-3, -2]``
+          (when ``[-3, -2]``, a channel first to channel last transpose would be inserted).
+        - ``gamma`` and ``beta`` are rank 1, after ``squeeze``.
+
+    It is ``layer_norm`` if all of the following are true:
+        - ``axes`` is either ``[-1]``, ``[-1, -2]``, or ``[-1, -2, -3]``, and so on.
+        - ``rank`` of ``gamma`` and ``beta`` is equal to the length of the ``axes``.
+    
+    `Pattern 2`
+        
+    Identify the pattern:
+        
+    ``y = (x - mean) / pow(variance + epsilon) * gamma + beta``
```

Comment:
I don't think you want a line break here after the plus sign.

Reply:
Now changed (removed). As per Aseem's suggestion, the reader can click the **[source]** button to see these docstrings in the source code. That way there are no duplicate docstrings.

Reply:
@TobyRoseman @aseemw : Changed to the following: "You can find these patterns in the methods for this class in the source code. To quickly view the source code, click the [source] button at the end of the class definition."